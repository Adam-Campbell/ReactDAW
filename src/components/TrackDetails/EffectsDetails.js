import React from 'react';
import PropTypes from 'prop-types';
import EffectDetailItem from './EffectDetailItem';
import { effectTypes } from '../../constants';

const EffectsDetails = props => (
    <div>
        <h2 className="track-details__heading">Effects</h2> 
        {props.effects.map(effect => (
            <EffectDetailItem 
                key={effect.id}
                effectType={effect.type}
                effectId={effect.id}
                handleOpen={props.handleOpen}
                handleRemove={props.handleRemove}
            />
        ))} 
        <div>
            <label 
                htmlFor="effect-select" 
                className="track-details__effect-select-label"
            >Add effect:</label>
            <select 
                id="effect-select" 
                className="track-details__effect-select"
                value={props.effectTypeToAdd}
                onChange={props.handleChange}
            >
                <option 
                    value={effectTypes.chorus}
                    className="track-details__effect-select-option"
                >Chorus</option>
                <option 
                    value={effectTypes.compressor}
                    className="track-details__effect-select-option"
                >Compressor</option>
                <option 
                    value={effectTypes.distortion}
                    className="track-details__effect-select-option"
                >Distortion</option>
                <option 
                    value={effectTypes.eq3}
                    className="track-details__effect-select-option"
                >EQ3</option>
                <option 
                    value={effectTypes.feedbackDelay}
                    className="track-details__effect-select-option"
                >Feedback Delay</option>
                <option 
                    value={effectTypes.filter}
                    className="track-details__effect-select-option"
                >Filter</option>
                <option 
                    value={effectTypes.freeverb}
                    className="track-details__effect-select-option"
                >Freeverb</option>
                <option 
                    value={effectTypes.gate}
                    className="track-details__effect-select-option"
                >Gate</option>
                <option 
                    value={effectTypes.jcReverb}
                    className="track-details__effect-select-option"
                >JC Reverb</option>
                <option 
                    value={effectTypes.limiter}
                    className="track-details__effect-select-option"
                >Limiter</option>
                <option 
                    value={effectTypes.phaser}
                    className="track-details__effect-select-option"
                >Phaser</option>
                <option 
                    value={effectTypes.tremolo}
                    className="track-details__effect-select-option"
                >Tremolo</option>
                <option 
                    value={effectTypes.vibrato}
                    className="track-details__effect-select-option"
                >Vibrato</option>
            </select>
            <button
                className="track-details__effect-button"
                onClick={() => props.handleAdd(props.effectTypeToAdd)}
            >Add</button>
        </div>
    </div>
);

EffectsDetails.propTypes = {
    handleOpen: PropTypes.func.isRequired,
    handleRemove: PropTypes.func.isRequired,
    handleAdd: PropTypes.func.isRequired,
    handleChange: PropTypes.func.isRequired,
    effects: PropTypes.arrayOf(PropTypes.object).isRequired,
    effectTypeToAdd: PropTypes.string.isRequired
};

export default EffectsDetails;