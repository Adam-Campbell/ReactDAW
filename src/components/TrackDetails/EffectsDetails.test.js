import React from 'react';
import { shallow } from 'enzyme';
import EffectsDetails from './EffectsDetails';

const effectsState = [
    {
        id: '3666421382061490',
        type: 'chorus',
        channelId: '3821569482884337',
        effectData: {
            delayTime: 3.5,
            depth: 0.7,
            frequency: 1.5,
            spread: 180,
            type: 'sine',
            wet: 1
        }
    }
];

test('renders correctly', () => {
    const component = shallow(
        <EffectsDetails 
            handleOpen={jest.fn()}
            handleRemove={jest.fn()}
            handleAdd={jest.fn()}
            handleChange={jest.fn()}
            effects={effectsState}
            effectTypeToAdd="chorus"
        />
    );
    expect(component).toMatchSnapshot();
});

