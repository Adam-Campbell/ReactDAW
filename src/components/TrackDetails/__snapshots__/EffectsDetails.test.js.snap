// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2
    className="track-details__heading"
  >
    Effects
  </h2>
  <EffectDetailItem
    effectId="3666421382061490"
    effectType="chorus"
    handleOpen={[MockFunction]}
    handleRemove={[MockFunction]}
    key="3666421382061490"
  />
  <div>
    <label
      className="track-details__effect-select-label"
      htmlFor="effect-select"
    >
      Add effect:
    </label>
    <select
      className="track-details__effect-select"
      id="effect-select"
      onChange={[MockFunction]}
      value="chorus"
    >
      <option
        className="track-details__effect-select-option"
        value="chorus"
      >
        Chorus
      </option>
      <option
        className="track-details__effect-select-option"
        value="compressor"
      >
        Compressor
      </option>
      <option
        className="track-details__effect-select-option"
        value="distortion"
      >
        Distortion
      </option>
      <option
        className="track-details__effect-select-option"
        value="eq3"
      >
        EQ3
      </option>
      <option
        className="track-details__effect-select-option"
        value="feedbackDelay"
      >
        Feedback Delay
      </option>
      <option
        className="track-details__effect-select-option"
        value="filter"
      >
        Filter
      </option>
      <option
        className="track-details__effect-select-option"
        value="freeverb"
      >
        Freeverb
      </option>
      <option
        className="track-details__effect-select-option"
        value="gate"
      >
        Gate
      </option>
      <option
        className="track-details__effect-select-option"
        value="jcReverb"
      >
        JC Reverb
      </option>
      <option
        className="track-details__effect-select-option"
        value="lfo"
      >
        LFO
      </option>
      <option
        className="track-details__effect-select-option"
        value="limiter"
      >
        Limiter
      </option>
      <option
        className="track-details__effect-select-option"
        value="phaser"
      >
        Phaser
      </option>
      <option
        className="track-details__effect-select-option"
        value="tremolo"
      >
        Tremolo
      </option>
      <option
        className="track-details__effect-select-option"
        value="vibrato"
      >
        Vibrato
      </option>
    </select>
    <button
      className="track-details__effect-button"
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</div>
`;
